react/GraphqlClient.tsx

```diff
@@ -44,6 +45,71 @@ export const getList = (
   })
 }
 
+const fetchListsByOwner = (
+  client: ApolloClient<ResponseList>,
+  owner: string,
+  page?: number,
+  pageSize?: number
+): Promise<ResponseList> =>
+  client.query({
+    fetchPolicy: 'network-only',
+    query: getListsByOwnerQuery,
+    variables: { owner, page, pageSize },
+  })
+
+const joinLists = (
+  listsByOwner: List[] | undefined,
+  listsNotIndexed: ResponseList[]
+) => {
+  const notIndexed = map(item => item.data.list || {}, listsNotIndexed)
+  const lists = filter(
+    item => contains(item.id, getListsIdFromCookies()),
+    listsByOwner || []
+  )
+  return concat(lists, notIndexed)
+}
+
+const isListFromOwner = (listsNotIndexed: ResponseList[], owner: string) => {
+  return (
+    listsNotIndexed.length > 0 &&
+    path(['data', 'list', 'owner'], listsNotIndexed[0]) !== owner
+  )
+}
+
+const getSyncLists = async (
+  client: ApolloClient<ResponseList>,
+  owner: string
+): Promise<ResponseList> => {
+  let {
+    data: { listsByOwner },
+  } = await fetchListsByOwner(client, owner)
+
+  const listsId = map(item => item.id, listsByOwner || [])
+  const listIdFromLocal = getListsIdFromCookies()
+  if (!listIdFromLocal || !listIdFromLocal.length) {
+    map(id => saveListIdInLocalStorage(id), listsId)
+  } else {
+    const listsIdNotIndexed = without(listsId, listIdFromLocal)
+    const listsNotIndexed = await Promise.all(
+      map(id => getList(client, id || ''), listsIdNotIndexed)
+    )
+
+    if (isListFromOwner(listsNotIndexed, owner)) {
+      localStorage.removeItem(WISHLIST_STORAGE_KEY)
+      map(id => saveListIdInLocalStorage(id), listsId)
```

Comment:
```suggestion
      map(saveListIdInLocalStorage, listsId)
```
will probably work just fine